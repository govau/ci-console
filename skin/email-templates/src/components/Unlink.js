import React from "react";
import PropTypes from "prop-types";

// Unlink attempts to trick mail clients that parse links and auto-wrapping them
// in anchor tags.
// Does not work in Apple Mail.
const Unlink = ({ children }) => {
  const parts = children.split(".");
  return parts.map((part, i) => {
    const separator = i < parts.length - 1 ? "." : "";
    return <span key={i}>{`${part}${separator}`}</span>;
  });
};

Unlink.propTypes = {
  children: PropTypes.string.isRequired
};

export default Unlink;
