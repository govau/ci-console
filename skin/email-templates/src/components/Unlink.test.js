import React from "react";
import renderer from "react-test-renderer";

import Unlink from "./Unlink";

test("renders without crashing", () => {
  const tree = renderer.create(<Unlink>www.example.com</Unlink>);
  expect(tree).toMatchSnapshot();
});
