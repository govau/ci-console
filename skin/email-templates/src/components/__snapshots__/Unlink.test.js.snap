// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <span>
    www.
</span>,
  <span>
    example.
</span>,
  <span>
    com
</span>,
]
`;
